core/src/main/java/com/linecorp/armeria/server/HttpServerHandler.java

```diff
@@ -445,10 +445,25 @@ private void handleRequest(ChannelHandlerContext ctx, DecodedHttpRequest req) th
         }
     }
 
-    private HttpResponse serve0(HttpRequest req,
-                                ServiceConfig serviceCfg,
-                                HttpService service,
-                                DefaultServiceRequestContext reqCtx) {
+    private static HttpResponse serve0(DecodedHttpRequest req, ServiceConfig serviceCfg, HttpService service,
+                                       DefaultServiceRequestContext reqCtx,
+                                       @Nullable EventLoop serviceEventLoop) {
+        final CompletableFuture<Void> whenAggregated = req.whenAggregated();
+        if (whenAggregated != null) {
+            return HttpResponse.of(whenAggregated.thenApply(ignored -> {
```

Comment:
Question) Since `whenAggregated` is completed from the channel event loop, wouldn't `serve1` be called from the `channelEventLoop` even though `serviceEventLoop` is non-null?

Reply:
It should be fixed.

Reply:
Added an additional method to optional use `serviceEventLoop` for the service.

Reply:
What do you think of just using if else?
```java
if (serviceEventLoop.inEventLoop()) {
    final CompletableFuture<Void> whenAggregated = req.whenAggregated();
    if (whenAggregated != null) {

    } else {

    }
} else {
    final CompletableFuture<Void> whenAggregated = req.whenAggregated();
    if (whenAggregated != null) {

    } else {

    }
}
```

Reply:
Changed the code snippet, PTAL again. 🙇

Reply:
`serviceEventLoop` is null if it is the same as `channelEventLoop`. Do you think that we should pass `serviceEventLoop` to `serve0()`?
```diff
- if (needsDirectExecution) {
-     res = serve0(req, serviceCfg, service, reqCtx, null);
- } else {
-     res = serve0(req, serviceCfg, service, reqCtx, serviceEventLoop);
- }
+ res = serve0(req, serviceCfg, service, reqCtx, serviceEventLoop);
```

Reply:
It looks good. Could you directly push the changes to this PR?

Reply:
Pushed. 😊

Reply:
Many thanks!